app/subcommands/deploy_subcommand.py

```diff
@@ -267,21 +267,21 @@ def _non_registered_slaves(self, registered_slaves, slaves_to_validate):
         :return: list of slave hostnames that haven't registered with the master service yet
         :rtype: list[str]
         """
-        registered_rsa_keys = []
+        registered_host_ids = []
 
         for registered_slave in registered_slaves:
-            registered_rsa_keys.append(Network.rsa_key(registered_slave))
+            registered_host_ids.append(Network.get_host_id(registered_slave))
```

Comment:
No. You're totally right. It was doing a little more. But now it makes perfect sense to use list comprehensions. I will update it. :smiley: